server/src/main/java/keywhiz/service/daos/ClientDAO.java

```diff
@@ -73,6 +74,18 @@ public void deleteClient(Client client) {
     });
   }
 
+  public void sawClient(Client client) {
+    long now = OffsetDateTime.now().toEpochSecond();
+
+    dslContext.transaction(configuration -> {
+      DSL.using(configuration)
+          .update(CLIENTS)
+          .set(CLIENTS.LASTSEEN, now)
```

Comment:
I would do this instead:
```
.set(CLIENTS.LASTSEEN, DSL.greatest(CLIENTS.LASTSEEN, DSL.val(now, CLIENTS.LASTSEEN)))
```

Which is jOOQ DSL for:
```
lastseen = greatest(lastseen, :now)
```

That way you avoid every accidentally writing an earlier value than what is already in the database in case of clock skew.